### ISSUE TITLE
New output items appended using the new API does not update the Notebook Document

### ISSUE THREAD

**Issue by DonJayamanne** [microsoft/vscode#116598]
- Version: 1.54.0-insider
- Commit: 6ac9a3ecb3698e82bf901f11bbb5940f6bc3c197
- Date: 2021-02-12T05:12:31.721Z
- Electron: 11.2.3
- OS: Darwin x64 20.3.0

**Steps to Reproduce:**

1. Use the new API and update an existing output by appending to it as follows:
```typescript
    edit.appendNotebookCellOutputItems(
        this.editor.document.uri,
        this.cell.index,
        existing.id,
        [.. new items .. ]
    );
```
2. THis works as expected, the UI is updated with the new output.
3. However when inspecting the Notebook Cells output (API), the new output item is not available.
NotebookDocument.cells[...].outputs[0].outputs <- this array doesn't get updated, only the UI gets updated.
I.e. it looks like VSCode is only updating the UI and not the underlying Notebook Cell output objects.

Unfortunately I cannot provide a sample code as this is still a WIP (we're updating our code to use the latest notebook API, its still only a change I have locally).
Oh yes, we need this information so that we can take the notebook and persist it to a file.

@jrieken @rebornix /cc

Does this issue occur when all extensions are disabled?: N/A

**Comment by jrieken** [comment]
> However when inspecting the Notebook Cells output (API), the new output item is not available.
NotebookDocument.cells[...].outputs[0].outputs <- this array doesn't get updated, only the UI gets updated.

Do have a full code sample for that?

**Comment by rebornix** [comment]
this is fixed via https://github.com/microsoft/vscode/issues/116711 @jrieken right?

**Comment by jrieken** [comment]
> this is fixed via #116711 @jrieken right?

I am pretty sure that's unrelated

**Comment by DonJayamanne** [comment]
> Do have a full code sample for that?

Again, the UI is updated, but looking at the cells.outputs.outputs, we only have the first entry in there.

Here's the code where we update existing output
https://github.com/microsoft/vscode-jupyter/blob/4f7f1d5641b8fc766c968d881ea8e1dfc91c5118/src/client/datascience/jupyter/kernels/cellExecution.ts#L663

Here are the tests for this scenario https://github.com/microsoft/vscode-jupyter/blob/301c4a95460e46f808bfbf7b296e6ab9542affd3/src/test/datascience/notebook/executionService.vscode.test.ts#L351
